src/widgets/sparkline.rs

```diff
@@ -127,6 +152,42 @@ impl<'a> Sparkline<'a> {
     }
 }
 
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+pub struct SparklineData(Vec<Option<u64>>);
```

Comment:
Non-blocking idea. This seems like it might be easy to extend to style individual data points in the future. Do you see any problem with this data type that would prevent us from making that sort of change (I don't but it's worth getting a second opinion on this). Are there any other obvious changes that we'd want to do here?

Reply:
I've added optional per-item styling.  

The styles cascade, the `Sparkline` stying will be used by default but can be overridden per item (i.e. `SparkItem`).  Note that empty item styling remains independent.

There are some news tests to demonstrate these "rainbow" Sparklines.